### PR TITLE
kubernetes: Fix hangs when connecting

### PR DESCRIPTION
**Pull request by petervo**
If a kube config has no user data for a context kubectl prompts. Unfortunately there is no way
to stop this. So only run kubectl version when when know we have a user with a auth-provider
section. Make sure this is the default user by using --minify to get the current user only.

### CODE REVIEW

File: pkg/kubernetes/scripts/kube-client-cockpit.js [hunk 1]
```diff
@@ -804,8 +804,9 @@
 
     .factory("cockpitKubectlConfig", [
         '$q',
+        "$timeout",
```
Comment (martinpitt) [inline pkg/kubernetes/scripts/kube-client-cockpit.js]: Incosistent quoting (but most likely irrelevant, see below).

File: pkg/kubernetes/scripts/kube-client-cockpit.js [hunk 2]
```diff
@@ -804,8 +804,9 @@
...
         'cockpitRunCommand',
-        function($q, runCommand) {
+        function($q, $timeout, runCommand) {
```
Comment (martinpitt) [inline pkg/kubernetes/scripts/kube-client-cockpit.js]: This isn't being used anywhere, did you mean to use it in the `kubectl version` call?

**Comment by martinpitt** [comment]
Can you please add a "Fixes #7688" to the commit log? (Or let's do it when merging, if the `$timeout` thing is intended as-is).

File: pkg/kubernetes/scripts/kube-client-cockpit.js (hunk 2 above)
Comment (petervo) [reply pkg/kubernetes/scripts/kube-client-cockpit.js]: Leftover cruft from my first approach. I'll remove.

**Pull request merged**